src/Sparrow.Server/ByteString.cs

```diff
@@ -1007,14 +1011,18 @@ public InternalScope AllocateDirect(int length, ByteStringType type, out ByteStr
             if (allocationUnit <= _internalCurrent.SizeLeft)
             {
                 var byteString = Create(_internalCurrent.Current, length, allocationUnit, type);
-                _internalCurrent.Current += byteString._pointer->Size;
+
+                allocationUnit = byteString._pointer->Size;
+                _internalCurrent.Current += allocationUnit;
+                _currentlyAllocated += allocationUnit;
```

Comment:
you sure? in `AllocateInternal` method we just do:

https://github.com/ravendb/ravendb/blob/62cbe67f2a6de184fbbf768cfde9870a9d4c740e/src/Sparrow.Server/ByteString.cs#L1124

Reply:
Yes, but that's is not the one we called. The idea of `AllocateDirect` is to avoid `AllocateInternal` because it is too costly for the high traffic sections. In the unlikely path we call `AllocateInternalUnlikely` instead which is the `AllocateInternal`s unlikely path too.